sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -51,6 +51,93 @@ APIs for managing module identities, module twins, and querying modules
 
 ```csharp
 
+public  class Modules
+{
+    /// <summary>
+    /// Create a device module.
+    /// </summary>
+    /// <param name="moduleIdentity">The module to create.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public virtual async Task<Response<ModuleIdentity>> CreateIdentityAsync(ModuleIdentity moduleIdentity, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Get a single device module.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device that contains the module.</param>
+    /// <param name="moduleId">The unique identifier of the module to get.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The retrieved device.</returns>
+    public virtual async Task<Response<ModuleIdentity>> GetIdentityAsync(string deviceId, string moduleId, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// List a set of device modules.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A pageable set of device modules.</returns>
+    public virtual async AsyncPageable<ModuleIdentity> GetIdentitiesAsync(string deviceId, CancellationToken cancellationToken = default);
```

Comment:
The service API that this method uses does not support pagination. This should be an IReadOnlyList<ModuleIdentity>